tests/fixture_valid.js

```diff
@@ -308,6 +311,35 @@ module.exports.checkFixture = function checkFixture(fixture, usedShortNames=[])
   return result;
 };
 
+function checkCoarseChannelExistence(result, fixture, fineChannels, switchingChannels, ch, mode, chNumber) {
+  const coarseChannelKey = fineChannels[ch];
+
+  // the mode must also contain the coarse channel
+  if (!mode.channels.includes(coarseChannelKey)) {
+    result.errors.push(`Mode '${mode.name || mode.shortName}' contains the fine channel '${ch}' (#${chNumber}) but is missing its coarse channel '${coarseChannelKey}'.`);
+  }
+
+  // the mode must also contain all coarser channels
+  const fineChannelAliases = fixture.availableChannels[coarseChannelKey].fineChannelAliases;
+  const coarserChannelKeys = fineChannelAliases.slice(0, fineChannelAliases.indexOf(ch));
+  for (const coarserChannelKey of coarserChannelKeys) {
+    // check if the coarse channel is used directly or as part of a switching channel
+    const isCoarseUsedInMode = mode.channels.some(chKey => {
+      // used in a switching channel
+      if (chKey in switchingChannels) {
+        const switchedChannels = switchingChannels[chKey].switchedChannels;
+        return switchedChannels.includes(coarserChannelKey)
+      }
+
+      // used directly
+      return chKey === coarserChannelKey;
```

Comment:
I'd swap the if/else blocks because this is the "normal" case and therefore should lead to an earlier result.